interview-questions.md

```diff
@@ -6,36 +6,38 @@ First, without external resources. Challenge yourself to answer from memory as i
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-Type your initials here to confirm you read the instructions:
+Type your initials here to confirm you read the instructions: MS
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is the description of the data, the argument is the actual data.
 
-Researched answer:
+Researched answer: A parameter is a variable or place holder in a function definition.
+An argument is a value passed to a function when it's called.
 
 2. What is the difference between an object and a class?
 
-Your answer:
+Your answer: Honestly I can put into words the exact difference.
 
-Researched answer:
+Researched answer: An object is an instance of a class that encapsulates data and behavior.
+A Class is aa blueprint/template for creating objects, defining their structure and behavior.
 
 3. What is iteration?
 
-Your answer:
+Your answer: Processed excuted over and over until a condition is met.
 
-Researched answer:
+Researched answer: The process of repeatedly executing a set of statements, typically to process each item in a collection or perform a specific task multiple times.
 
-4. STRETCH: What is hoisting in JavaScript?
+4. STRETCH: What is hoisting in JavaScript? 
 
-Answer:
+Answer: I had to research this, because I didn't know. Hoisting is a behavior in JavaScript where the variable and the function declarations are moved to the top of their containing scope during the compilation phase, regardless of where they are declared in the code. This allows variables and functions to be used before they are declared. But, only the declarations are hoisted, not the initializations or assignments.
 
 ## Looking Ahead: Terms for Next Unit
 
 Research and define the following terms. Hint: use the syllabus!
 
-1. React State:
+1. React State: There is a specific React data structure called state that is used to store data within a React component. Data stored in state is available to the entire React component.
 
-2. React Props:
+2. React Props: Props, short for properties, is a specialized tool in React for the communication of data and behavior between components.
 
-3. DOM events:
+3. DOM events: The DOM (Document Object Model) is always listening for changes.  These interactions can happen in a variety of ways. The user can click with their mouse, hover over elements, or use their keyboard to type. All of these interactions are considered DOM events.
```

Comment:
great job with the research.  I feel you can push yourself in these interview questions.  Remember, pretend you're in an interview -- how can you best show them you understand the concept.  Keep grinding Mark.  you're doing great!